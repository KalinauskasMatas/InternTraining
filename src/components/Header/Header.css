.header-name {
  color: var(--bright-text-color);
  transform: skew(-10deg);
  font-size: 27px;
  margin-top: 22px;
  margin-bottom: 30px;
  padding-left: 6px;
  letter-spacing: 2px;
  word-spacing: -6px;
}