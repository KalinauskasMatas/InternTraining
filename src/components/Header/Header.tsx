import "./Header.css";
import { Title } from "../../interfaces";

const Header = (props: Title) => {
  return (
    <header className="header-name" data-testid="header-title">
      {props.title}
    </header>
  );
};

export default Header;
