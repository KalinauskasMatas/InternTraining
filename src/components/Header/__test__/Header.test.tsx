import Header from "../Header";

import { render, screen, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";

import renderer from "react-test-renderer";

afterEach(cleanup);

it("renders without crashing", () => {
  render(<Header title="Title" />);
});

it("renders header correctly", () => {
  render(<Header title={"Title"} />);
  expect(screen.getByTestId("header-title")).toHaveTextContent("Title");
});

it("renders header correctly 2", () => {
  render(<Header title={"title-2"} />);
  expect(screen.getByTestId("header-title")).toHaveTextContent("title-2");
});

it("matches snapshot", () => {
  const tree = renderer.create(<Header title={"Title"} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("matches snapshot 2", () => {
  const tree = renderer.create(<Header title={"Other title"} />).toJSON();
  expect(tree).toMatchSnapshot();
});
