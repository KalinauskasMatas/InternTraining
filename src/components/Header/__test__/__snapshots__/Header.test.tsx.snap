// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<header
  className="header-name"
  data-testid="header-title"
>
  Title
</header>
`;

exports[`matches snapshot 2 1`] = `
<header
  className="header-name"
  data-testid="header-title"
>
  Other title
</header>
`;
